README.md

```diff
@@ -0,0 +1,105 @@
+directly harm others. HelenOS is therefore flexible, modular,
+extensible, fault tolerant and easy to understand.
+
+HelenOS does not aim to be a clone of any existing operating system
+and trades compatibility with legacy APIs for cleaner design.
+Most of HelenOS components have been made to order specifically for
+HelenOS so that its essential parts can stay free of adaptation layers,
+glue code, franken-components and the maintenance burden incurred by them.
+
+* [Website](http://helenos.org)
+* [Wiki](http://helenos.org/wiki)
+* [Tickets](http://www.helenos.org/report/1)
+* [How to contribute](http://www.helenos.org/wiki/HowToContribute)
+
+## Portability
+
+HelenOS runs on seven different processor architectures and machines ranging
+from embedded ARM devices and single-board computers through multicore 32-bit
+and 64-bit desktop PCs to 64-bit Itanium and SPARC rack-mount servers.
+
+## Building
+
+### Building the toolchain
+
+In order to build HelenOS, one must first build the cross-compiler toolchain
+(either as a root or by specifying the `CROSS_PREFIX` environment variable)
+by running (example for the amd64 architecture, further list of targets can be
+found in the `default` directory):
+
+```
+$ cd HelenOS/tools
+$ ./toolchain.sh amd64
+```
+
+The toolchain script will print a list of software packages that are required
+for the toolchain to correctly build. Make sure you install all the dependencies.
+Unfortunately, the script cannot install the required dependencies for you automatically
+since the host environments are very diverse. In case the compilation of the toolchain
+fails half way through, try to analyze the error message(s), add appropriate missing
+dependencies and try again.
+
+As an example, here are some of the packages you will need for Ubuntu 12.10 (may be out of date):
+
+```
+$ sudo apt-get install build-essential libgmp-dev libmpfr-dev ppl-dev libmpc-dev zlib1g-dev texinfo libtinfo-dev xutils-dev
+```
+
+Whereas for CentOS/Fedora, you will need:
+
+```
+# sudo dnf group install 'Development Tools'
+# sudo dnf install wget texinfo libmpc-devel mpfr-devel gmp-devel PyYAML genisoimage
+```
+In case the toolchain script won't work no matter how hard you try, let us know.
+Please supply as many relevant information (your OS and distribution, list of
+installed packages with version information, the output of the toolchain script, etc.) as
+possible.
+
+### Configuring the build
+
+Go back to the source root of HelenOS and start the build process:
+
+```
+$ cd ..
+$ make PROFILE=amd64
+```
+
+Now HelenOS should automatically start building.
+
+Note: If you installed the toolchain to a custom directory, make sure `CROSS_PREFIX`
+environment variable is correctly set.
+
+### Running the OS
+
+When you get the command line back, there should be an `image.iso` file in the source
+root directory. If you have QEMU, you should be able to start HelenOS by running:
+
+```
+$ qemu-system-i386 -m <t_co> -boot d -cdrom image.iso
```

Comment:
Should be `-x86_64`?

Reply:
I'd say that under all circumstances, the user should run tools/ew.py.

Reply:
Yeah.

Reply:
Yes, I missed that. Also, the <t_co> after -m wierds me out, that's supposed to be 256. Now that I went to fix it in the file, it seems that anytime I write 256 in vim it gets changed to 256 :D

Reply:
> anytime I write 256 in vim it gets changed to 256
Not entirely surprising. :D

Reply:
... Meant <t_co> :)

Reply:
Fixed.